### PR TITLE
Make MiniEngine core compatible with /W4/WX

### PR DESCRIPTION
**Pull request by RobJenks**
Address all warnings to allow compilation of all MiniEngine components under /W4/WX.

I have tried to make changes as minimal as possible and there are no functional effects.  Commits split by warning type to make the changes a little clearer

### CODE REVIEW

**Comment by RobJenks** [comment]
Reference issue #492

File: MiniEngine/Tools/SDFFontCreator/SDFFontCreator.cpp [hunk 1]
```diff
@@ -445,8 +445,8 @@ void CompileFont(const string& inputFile, uint32_t size, const string& outputNam
             if (wc >= L'0' && wc <= L'9')
             {
                 int extraSpace = numberWidth - g_glyphs[i].width;
-                g_glyphs[i].bearing = extraSpace / 2;
-                g_glyphs[i].advance = numberWidth;
+				g_glyphs[i].bearing = static_cast<uint16_t>(extraSpace) / 2U;
+				g_glyphs[i].advance = static_cast<uint16_t>(numberWidth);
```
[the remark this answers is not in the document]
Comment (RobJenks) [reply MiniEngine/Tools/SDFFontCreator/SDFFontCreator.cpp]: I'll address this now (and the couple of other instances) - thanks

[the remark this answers is not in the document]
Comment (RobJenks) [reply MiniEngine/Tools/SDFFontCreator/SDFFontCreator.cpp]: This was addressed by commit 937199b

**Comment by RobJenks** [comment]
The commits in this PR should resolve all level 4 warnings for the MiniEngine projects and tools:
* MiniEngine/Core
* MiniEngine/Model
* MiniEngine/ModelViewer
* MiniEngine/ModelConverter
* MiniEngine/Tools/SDFFontCreator
* The new root project generated by CreateNewSolution.bat

I haven't updated the vcproj configuration from /W3 to /W4 in this PR.  If you're happy to do so (it looks like many of the MS projects come with this as default) I'll raise it as a follow-up PR.

**Comment by stanard** [comment]
I went through and made changes locally and pushed them up.  It's pretty similar to what Rob did but a few differences.  We were already trying to disable warning 4238 'nonstandard extension used : class rvalue used as lvalue', but there was a typo calling it 4**32**8.  Whether it is nonstandard is irrelevant because it works in our chosen compiler.  I also think static_cast is unnecessarily verbose when you can use a C-style cast.  I updated SDFFontGenerator, but I didn't update ModelConverter.  ModelConverter will eventually be replaced by a lazy content compiler system when I push out glTF 2.0 model format support.

**Pull request closed without merging**